sycl/test-e2e/Graph/Inputs/interop-level-zero-launch-kernel.cpp

```diff
@@ -0,0 +1,169 @@
+  auto NodeA = add_node(
+      Graph, Queue, [&](handler &CGH) { CGH.copy(HostZ.data(), MemZ, Size); });
+
+  auto NodeB = add_node(
+      Graph, Queue, [&](handler &CGH) { CGH.copy(HostX.data(), MemX, Size); });
+
+  auto NodeC = add_node(
+      Graph, Queue,
+      [&](handler &CGH) {
+        depends_on_helper(CGH, {NodeA, NodeB});
+        CGH.host_task([&]() {
+          auto ZeContext = get_native<backend::ext_oneapi_level_zero>(Context);
+          auto ZeDevice = get_native<backend::ext_oneapi_level_zero>(Device);
+
+          ze_result_t status;
+          ze_module_desc_t moduleDesc = {ZE_STRUCTURE_TYPE_MODULE_DESC,
+                                         nullptr,
+                                         ZE_MODULE_FORMAT_IL_SPIRV,
+                                         Spirv.size(),
+                                         Spirv.data(),
+                                         nullptr,
+                                         nullptr};
+          ze_module_handle_t hModule;
+          status = zeModuleCreate(ZeContext, ZeDevice, &moduleDesc, &hModule,
+                                  nullptr);
+          assert(status == ZE_RESULT_SUCCESS);
+
+          ze_kernel_desc_t kernelDesc = {
+              ZE_STRUCTURE_TYPE_KERNEL_DESC, nullptr, 0,
+              "_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E5saxpy"};
+          ze_kernel_handle_t hKernel;
+          status = zeKernelCreate(hModule, &kernelDesc, &hKernel);
+          assert(status == ZE_RESULT_SUCCESS);
+
+          auto ZeCommandQueueDesc =
+              ze_command_queue_desc_t{ZE_STRUCTURE_TYPE_COMMAND_QUEUE_DESC,
+                                      nullptr,
+                                      0,
+                                      0,
+                                      0,
+                                      ZE_COMMAND_QUEUE_MODE_SYNCHRONOUS,
+                                      ZE_COMMAND_QUEUE_PRIORITY_NORMAL};
+
+          ze_command_list_handle_t ZeCommandList;
+          status = zeCommandListCreateImmediate(
+              ZeContext, ZeDevice, &ZeCommandQueueDesc, &ZeCommandList);
+          assert(status == ZE_RESULT_SUCCESS);
+
+          status = zeKernelSetArgumentValue(hKernel, 0, Size * sizeof(uint32_t),
+                                            &MemZ);
+          assert(status == ZE_RESULT_SUCCESS);
+          status = zeKernelSetArgumentValue(hKernel, 1, Size * sizeof(uint32_t),
+                                            &MemX);
+          assert(status == ZE_RESULT_SUCCESS);
+          ze_group_count_t ZeGroupCount{Size, 1, 1};
+
+          zeKernelSetGroupSize(hKernel, 1024, 1, 1);
+          assert(status == ZE_RESULT_SUCCESS);
+
+          status = zeCommandListAppendLaunchKernel(
+              ZeCommandList, hKernel, &ZeGroupCount, nullptr, 0, nullptr);
+
+          assert(status == ZE_RESULT_SUCCESS);
+
+          status = zeCommandListHostSynchronize(ZeCommandList, 0);
+          assert(status == ZE_RESULT_SUCCESS);
```

Comment:
After this synchronize can we destroy the L0 objects we created? i.e. the command-list, kernel, and module